File: ParentLauncherApp/src/com/example/parentlauncher/MainActivity.java
package com.example.parentlauncher;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Toast;

import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends Activity {

    private static final int REQUEST_CODE_CREATE_CHILD_USER = 1;
    private MainViewModel mainViewModel;

    private Button createChildButton;
    private Button launchChildAppButton;
    private Button manageRestrictionsButton;
    private Button listUsersButton;
    private ImageButton backButton;
    private ImageButton settingsButton;
    private ImageButton homeButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainViewModel = new ViewModelProvider(this).get(MainViewModel.class);

        initializeViews();
        setupListeners();

        mainViewModel.getCurrentChildUserId().observe(this, userId -> {
            if (userId != -1) {
                Toast.makeText(this, "Last active child user ID loaded: " + userId, Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void initializeViews() {
        createChildButton = findViewById(R.id.create_child_button);
        launchChildAppButton = findViewById(R.id.launch_child_app);
        manageRestrictionsButton = findViewById(R.id.manage_restrictions_button);
        listUsersButton = findViewById(R.id.list_users_button);
        backButton = findViewById(R.id.back_button);
        settingsButton = findViewById(R.id.settings_button);
        homeButton = findViewById(R.id.home_button);
    }

    private void setupListeners() {
        createChildButton.setOnClickListener(v -> openCreateChildUserScreen());
        launchChildAppButton.setOnClickListener(v -> mainViewModel.launchChildApp(mainViewModel.getCurrentChildUserId().getValue()));
        manageRestrictionsButton.setOnClickListener(v -> mainViewModel.openRestrictionSettings(mainViewModel.getCurrentChildUserId().getValue()));
        listUsersButton.setOnClickListener(v -> openListUsersScreen());
        backButton.setOnClickListener(v -> onBackPressed());
        settingsButton.setOnClickListener(v -> mainViewModel.launchSystemSettings());
        homeButton.setOnClickListener(v -> mainViewModel.launchHomeScreen());
    }

    private void openCreateChildUserScreen() {
        Intent intent = new Intent(this, CreateChildUserActivity.class);
        startActivityForResult(intent, REQUEST_CODE_CREATE_CHILD_USER);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_CODE_CREATE_CHILD_USER) {
            if (resultCode == RESULT_OK) {
                if (data != null) {
                    int newChildUserId = data.getIntExtra(CreateChildUserActivity.EXTRA_CHILD_USER_ID, -1);
                    String profileType = data.getStringExtra(CreateChildUserActivity.EXTRA_PROFILE_TYPE);
                    if (newChildUserId != -1 && profileType != null) {
                        mainViewModel.saveCurrentChildUserId(newChildUserId);
                        Toast.makeText(this, "Child user created and selected: ID " + newChildUserId + ", Type: " + profileType, Toast.LENGTH_SHORT).show();
                    } else {
                        Toast.makeText(this, "Failed to get new child user ID.", Toast.LENGTH_SHORT).show();
                    }
                }
            } else if (resultCode == RESULT_CANCELED) {
                Toast.makeText(this, "Child user creation cancelled.", Toast.LENGTH_SHORT).show();
            }
        }
    }

    private void openListUsersScreen() {
        Intent intent = new Intent(this, UserListActivity.class);
        startActivity(intent);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }
}

File: BoyChildApplication/src/com/example/boyprofile/view/MainActivity.kt
package com.example.boyprofile.view

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import com.example.boyprofile.R

class MainActivity : AppCompatActivity() {

    private var childUserId: Int = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Retrieve child user ID from Intent
        childUserId = intent.getIntExtra("child_user_id", -1)

        // Set up category buttons
        val btnWatch = findViewById<Button>(R.id.btnWatch)
        val btnPlay = findViewById<Button>(R.id.btnPlay)
        val btnDraw = findViewById<Button>(R.id.btnDraw)
        val btnRead = findViewById<Button>(R.id.btnRead)

        btnWatch.setOnClickListener { launchCategory("watch") }
        btnPlay.setOnClickListener { launchCategory("play") }
        btnDraw.setOnClickListener { launchCategory("draw") }
        btnRead.setOnClickListener { launchCategory("read") }

        // Optional: Exit button
        val btnExit = findViewById<ImageButton>(R.id.exit_button)
        btnExit?.setOnClickListener {
            finish()
        }
    }

    private fun launchCategory(category: String) {
        val intent = Intent(this, AppGridActivity::class.java).apply {
            putExtra("category", category)
            putExtra("child_user_id", childUserId)
        }
        startActivity(intent)
    }
}

